src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,275 @@
+		  logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Failed Tests: "+Integer.toString(failedNumberOfTests));
+		}
+    catch (Exception e){
+    	logger.log(ModuleID.TEST, LogLevel.WARNING, "Exception in module: "+strModule+" >> "+e);      	
+    }
+  }
+
+
+ /**
+  * Void method to run test case of a given priority
+  * @param priority The priority level of the test to be run
+  */
+  public void runByPriority(int priority){
+    int totalNumberOfTests = 0;
+    int successfulNumberOfTests = 0;
+    int failedNumberOfTests = 0;
+    
+    try{
+		  ILogger logger = LoggerFactory.getLoggerInstance();
+		  
+		  //get list of all the tests using helper static method getAllTests()
+		  ArrayList<File> allTests = getAllTests();
+		  totalNumberOfTests = allTests.size();
+		  
+		  for (File testFile : allTests){
+		  	//Get full qualified class name from absolute path of testcase
+			  String absPath =  testFile.getAbsolutePath();
+			 	String[] arrOfStr = absPath.split("src/", 2); 
+			 	arrOfStr = arrOfStr[1].split(".java", 2);
+			 	String relPath = arrOfStr[0];
+			 	String fullQualifiedClassName = relPath.replace("/",".");
+
+		    Class<?> testClass = Class.forName(fullQualifiedClassName);
+		    Object test = testClass.getDeclaredConstructor().newInstance();
+		    
+		    if(priority == (test.getPriority())){
+		       boolean result = test.run(); 
+		       
+		       if(result == false){
+		        failedNumberOfTests++;
+		        if(failedNumberOfTests == 1){
+		          logger.log(ModuleID.TEST, LogLevel.INFO, "Failed Tests...");
+		        }
+		        //log the result of failed test cases
+		        logger.log(ModuleID.TEST, LogLevel.INFO, Integer.toString(failedNumberOfTests)+". "+testName+": "+test.getError());
+		       }
+		       else{
+		          successfulNumberOfTests++;
+		       }
+		      
+		    }  
+		  } 
+		    
+		  //result logging
+		  logger.log(ModuleID.TEST, LogLevel.INFO, "\nOverall Result:");
+		  logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Tests: "+Integer.toString(totalNumberOfTests));
+		  logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Successful Tests: "+Integer.toString(successfulNumberOfTests));
+		 	logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Failed Tests: "+Integer.toString(failedNumberOfTests));
+		 }
+		catch (Exception e){
+    	logger.log(ModuleID.TEST, LogLevel.WARNING, "Exception >> "+e);      	
+    }
+  }
+
+
+ /**
+  * Run all test cases of all modules one by one 
+  */
+  public void runAll(){
+    int totalNumberOfTests = 0;
+    int successfulNumberOfTests = 0;
+    int failedNumberOfTests = 0;
+    
+    try{
+		  ILogger logger = LoggerFactory.getLoggerInstance();
+		   
+		  //call the helper method to get list of all test cases 
+		  ArrayList<File> allTests = getAllTests();
+		  totalNumberOfTests = allTests.size();
```

Comment:
Is this much indentation needed? It should match with the code inside catch block